components/query/query-vector.service.js

```diff
@@ -82,6 +81,38 @@ export default function (
     }
     me.createFeatureAttributeList();
   });
+  $rootScope.$on(
+    'vectorQuery.featureSelected',
+    HsUtilsService.debounce(
+      (event, feature) => {
+        const layer = feature.getLayer(HsMapService.map);
+        if (angular.isDefined(layer) && layer.get('onFeatureSelected')) {
+          const originalFeature = me.getSelectedFeature(feature);
+          if (originalFeature) {
+            layer.get('onFeatureSelected')(originalFeature);
+          }
+        }
+      },
+      150,
+      false,
+      me
+    )
+  );
+  me.getSelectedFeature = function (feature) {
+    let original;
+    if (angular.isDefined(feature)) {
+      original = feature;
+      if (original.get('features') && original.get('features').length == 1) {
+        original = original
+          .get('features')
+          .map((feature) => feature)
+          .pop();
```

Comment:
```suggestion
        original = original.get('features').slice(-1)[0];
```

It could be some millisecond faster...

Reply:
Okey, will replace it like so!

Reply:
Since you know that the length == 1 then original.get('features')[0] should be enough

Reply:
That is very correct assumption @raitisbe , how could we have missed that?